google/cloud/bigtable/async_operation.h

```diff
@@ -0,0 +1,74 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGTABLE_ASYNC_OPERATION_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGTABLE_ASYNC_OPERATION_H_
+
+#include "google/cloud/bigtable/version.h"
+
+namespace google {
+namespace cloud {
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+namespace internal {
+class CompletionQueueImpl;
+}  // namespace internal
+
+/**
+ * Represents a pending asynchronous operation.
+ *
+ * When applications create an asynchronous operations with a `CompletionQueue`
+ * they provide a callback to be invoked when the operation completes
+ * (successfully or not). The completion queue type-erases the callback and
+ * hides it in a class derived from `AsyncOperation`.  A shared pointer to the
+ * `AsyncOperation` is returned by the completion queue so library developers
+ * can cancel the operation if needed.
+ */
+class AsyncOperation {
+ public:
+  virtual ~AsyncOperation() = default;
+
+  /**
+   * Requests that the operation be canceled.
+   *
+   * The result of canceling the operation is reported via the callback
+   * registered when the operation was created.
+   */
+  virtual void Cancel() = 0;
```

Comment:
No status returned, not even a boolean? Nothing?

Reply:
As the comment says, there is nothing to report, you request a cancel and that is it.  The result (whether the cancel goes through or not) is reported via the completion callback.